src/asynqp/routing.py

```diff
@@ -33,22 +37,27 @@ class Sender(object):
     def __init__(self, channel_id, protocol):
         self.channel_id = channel_id
         self.protocol = protocol
+        self.connection_exc = None
 
     def send_method(self, method):
+        if self.connection_exc is not None:
+            raise self.connection_exc
         self.protocol.send_method(self.channel_id, method)
 
+    def killall(self, exc):
+        """ Connection/Channel was closed. All subsequent requests should
+            raise an error
+        """
+        self._connection_exc = exc
```

Comment:
I will remove it from this PR, and create a separate issue, so we can return to it later.